### PR TITLE
feat: Add image display tests and enhance QuickFix UI components

### LINKED ISSUE #251: Add Images to the message screen

### PR DESCRIPTION
**Pull request by daferya**
This pull request includes several changes to the Android test and UI codebase, focusing on improving test coverage, updating UI elements, and refining the `QuickFixDetailsScreen` component. The most important changes include adding a new test class, modifying existing tests, and refining the UI layout and logic.

### Test Enhancements:

* **New Test Class**: Added `QuickFixDisplayImagesScreenTest` to verify the correct display of images and navigation actions.
* **Modified Tests**: Updated `QuickFixDetailsUserNoModeScreenTest` to use `onNodeWithTag("imageCard")` instead of `onAllNodesWithText("Image")` for image placeholders.

### UI Refinements:

* **QuickFixDetailsScreen**: Refined the `QuickFixDetailsScreen` component to improve layout and logic, including better handling of image displays and service listings. [[1]](diffhunk://#diff-867ef3f5d07fba01eb682a99b2e41b5006460a3d5ef4de1028e7e9fc9ca60958L4-R5) [[2]](diffhunk://#diff-867ef3f5d07fba01eb682a99b2e41b5006460a3d5ef4de1028e7e9fc9ca60958R18-R35) [[3]](diffhunk://#diff-867ef3f5d07fba01eb682a99b2e41b5006460a3d5ef4de1028e7e9fc9ca60958L46-R48) [[4]](diffhunk://#diff-867ef3f5d07fba01eb682a99b2e41b5006460a3d5ef4de1028e7e9fc9ca60958R58-R84) [[5]](diffhunk://#diff-867ef3f5d07fba01eb682a99b2e41b5006460a3d5ef4de1028e7e9fc9ca60958L102-R121) [[6]](diffhunk://#diff-867ef3f5d07fba01eb682a99b2e41b5006460a3d5ef4de1028e7e9fc9ca60958L129-R212)

### Codebase Improvements:

* **Navigation Actions**: Added `NavigationActions` to `QuickFixSlidingWindowContentTest` and updated related test methods to include navigation actions. [[1]](diffhunk://#diff-f90af2665e9dd1edaeafdb06c94e8958f4c62141a5db9dc5836d2b4c2415ba11R11-R13) [[2]](diffhunk://#diff-f90af2665e9dd1edaeafdb06c94e8958f4c62141a5db9dc5836d2b4c2415ba11R55-R66) [[3]](diffhunk://#diff-f90af2665e9dd1edaeafdb06c94e8958f4c62141a5db9dc5836d2b4c2415ba11L66-R76) [[4]](diffhunk://#diff-f90af2665e9dd1edaeafdb06c94e8958f4c62141a5db9dc5836d2b4c2415ba11L80-R91)

These changes collectively improve the test coverage, enhance the user interface, and ensure better maintainability of the codebase. 
Capture d’écran 2024-12-17 190253
Capture d’écran 2024-12-17 190307

### CODE REVIEW

File: app/src/androidTest/java/com/arygm/quickfix/ui/elements/QuickFixDisplayImagesScreenTest.kt
Comment (mohagpt) [inline app/src/androidTest/java/com/arygm/quickfix/ui/elements/QuickFixDisplayImagesScreenTest.kt:1]: This test file looks really solid! It covers a wide range of scenarios and edge cases, ensuring robust coverage. One small suggestion: please keep all inline comments in English so the repository’s documentation remains consistent. Otherwise, this test is well-structured and thorough.

File: app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixDetailsdescription.kt [hunk 1]
```diff
@@ -99,58 +100,116 @@ fun QuickFixDetailsScreen(
...
-                      style = poppinsTypography.labelMedium,
-                      color = MaterialTheme.colorScheme.primary)
-                }
-          }
+              modifier = Modifier.padding(bottom = 8.dp))
+          // Show the full description or truncate it based on isExpanded
+          Text(
+              text =
+                  if (isExpanded) quickFix.title
+                  else quickFix.title.take(250) + if (quickFix.title.length > 250) "..." else "",
```
Comment (mohagpt) [inline app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixDetailsdescription.kt:114]: One possible suggestion here would be to specify the choice of 250, maybe by creating a specific constant.

**Review by mohagpt — DISMISSED** [review]
This is a great enhancement! The code is clean, well-structured, and overall looks good. I’ve added two suggestions that you might want to consider taking into account in a future PR to further improve consistency and maintainability. Keep up the great work!

File: app/src/androidTest/java/com/arygm/quickfix/ui/elements/QuickFixDisplayImagesScreenTest.kt
Comment (mohagpt) [reply app/src/androidTest/java/com/arygm/quickfix/ui/elements/QuickFixDisplayImagesScreenTest.kt:1]: This change isn't needed to merge into main, but make sure to modify this for the next pr.

Comment (daferya) [reply app/src/androidTest/java/com/arygm/quickfix/ui/elements/QuickFixDisplayImagesScreenTest.kt:1]: thank you for your review I will change it

File: app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixDetailsdescription.kt (hunk 1 above)
Comment (daferya) [reply app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixDetailsdescription.kt:114]: It was already decided with team to take the value 250

**Review by mohagpt — APPROVED** [review]
The PR remains correct as only comments have been modified. Thanks for your reactivity and quick updates! 👍
